From 550376ce0613b997c85dfb700793843fdf66f6bc Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Wed, 2 Mar 2022 14:18:27 +0100
Subject: [PATCH 11/38] fastboot: Fallback to "raw" partition type if fastboot
 hal isn't present

Fastboot format fails to wipe any partition that doesn't at least return
"raw" partition type. Also both android.hardware.fastboot@1.0-impl.pixel
and android.hardware.fastboot@1.1-impl-mock return FileSystemType::RAW
so I assume this is fine.

Change-Id: I5707bddb1ba32edb6359858853d7b1afbf138b9f
---
 fastboot/device/variables.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fastboot/device/variables.cpp b/fastboot/device/variables.cpp
index ee1eed876..bb54bf2a3 100644
--- a/fastboot/device/variables.cpp
+++ b/fastboot/device/variables.cpp
@@ -339,8 +339,8 @@ bool GetPartitionType(FastbootDevice* device, const std::vector<std::string>& ar
 
     auto fastboot_hal = device->fastboot_hal();
     if (!fastboot_hal) {
-        *message = "Fastboot HAL not found";
-        return false;
+        *message = "raw";
+        return true;
     }
 
     FileSystemType type;
-- 
2.34.1

